### PR TITLE
Update for tooling in support of KubeVirt v 1.0 / Docs revamp

### LINKED ISSUE #576: Broken links to the user guide are not checked
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:

/kind bug

**What happened**:

Some blog posts contain links to the user guide that are broken.

The link checker does not verify them because the external checker does not check kubevirt.io URLs and the internal checker only checks website pages.

**What you expected to happen**:

- [ ] No broken links
- [ ] Link checker also verifies links to the user guide

**Anything else we need to know?**:

> URL where the problem can be found ...
> If the issue is with a lab, please provide information about your environment, platform, versions, ...

Examples include:

- https://kubevirt.io/user-guide/#/workloads/virtual-machines/assigning-vms-to-nodes?id=affinity-and-anti-affinity
- https://kubevirt.io/user-guide/#/workloads/virtual-machines/virtualized-hardware-configuration?id=clock
- https://kubevirt.io/user-guide/#/workloads/virtual-machines/disks-and-volumes

### LINKED ISSUE #634: Spell checker not checking html files
/kind bug

**What happened**:
Spell checker not checking html files.  This opens us up to spelling mistakes.

I suggest we add this additional command to `.travis.yml`
`- yaspeller --only-errors ./**/*.html`

### PR DESCRIPTION
**Pull request by mazzystr**
**What this PR does / why we need it**:
Re Makefile ... See user-guide [comment](https://github.com/kubevirt/user-guide/issues/359#issuecomment-700810450) for list on what this Makefile needs to do
RE casperjs test script ... this script takes in a url and file csv pair and tests the presence of the selector
RE Rakefile ... Update to add test cases for website to userguide links
RE 2019-07-08-kubevirt-with-ansible-part-2.markdown ... Fixed the broken links that have been hiding in this post

**Does this PR fix any issue?** 
> Fixes #634 
> Fixes #576 

**NOTES:**
* See user-guide [issues/#359](https://github.com/kubevirt/user-guide/issues/359) for notes driving this pull request.
*  FUNCTIONALITY IN MAKEFILE REQS [project-infra/pull/673](https://github.com/kubevirt/project-infra/pull/673)
* THIS PR WILL NOT AFFECT ANY EXISTING FUNCTIONALITY

### CODE REVIEW

**Comment by mazzystr** [comment]
This PR is ready for reviewing

@fabiand @rmohr  I would love input is you have minutes to spare

**Comment by rmohr** [comment]
@mazzystr looks good to me in general. I guess it would make sense if @codificat reviews and merges this. Thanks for the work.

**Review by codificat — CHANGES_REQUESTED** [review]
Reviewed/tested live and some issues were found with container permissions in the run-* targets.

Besides these, just one minor comment so far

**Comment by codificat** [comment]
I tried to run `make checks` ane I'm getting some permission errors, e.g.

```sh
$ make check_spelling_kubevirtio

Makefile: Check kubevirtio spelling
podman run -it --rm --name yaspeller --net=host -v /scratch/git/kubevirt/website:/srv"" -v /scratch/git/kubevirt/project-infra/images/yaspeller/.yaspeller.json:/srv/.yaspeller.json"" yaspeller /bin/bash -c "echo; yaspeller -c /srv/.yaspeller.json --only-errors --ignore-tags iframe,img,code,kbd,object,samp,script,style,var /srv" | \
sed -e 's/\/srv/\./g'

Error: EACCES: permission denied, open './.yaspeller.json'
```

This seems to be SELinux:

```sh
$ ausearch -m avc -ts recent
----
time->Mon Nov  2 20:20:35 2020
type=AVC msg=audit(1604344835.587:339): avc:  denied  { read } for  pid=22903 comm="node" name=".yaspeller.json" dev="dm-0" ino=1701018 scontext=system_u:system_r:container_t:s0:c329,c623 tcontext=unconfined_u:object_r:default_t:s0 tclass=file permissive=0
```

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ language: ruby
 cache:
   bundler: true
 rvm:
-- 2.5.3
+- 2.5.8
 before_install:
 - bundle install
 - npm install -g yaspeller
```
[the remark this answers is not in the document]
Comment (mazzystr) [reply .travis.yml:8]: I would love too update Travis to use the new container but just not right this sec.  We need to crawl and walk before running.  I was having a problem with rake and ruby maps and thought the version mismatch might have been behind the it.  It wasn't.  It's just ruby doing what ruby does.

It's a nonchanging change.  I'm happy to back this diff out.

**Comment by mazzystr** [comment]
> ```shell
> -v /scratch/git/kubevirt/website:/srv""
> ```

notice...
`-v /scratch/git/kubevirt/website:/srv""`

SELinux is not set 

Try...
`export SELINUX_ENABLED=True; make check_links_website`

Also note that I removed the `checks` target to clarify functionality.

**Comment by mazzystr** [comment]
We're getting close to this pull being mergable.  Three more items need to be addressed....

* spellchecker ... spellchecker has empty output on success.  I need to add a complete! msg
* spellchecker .... retval is always 0 due to `cmd | sed` and `echo`.  The echo can be removed since it's cosmetic.  The sed is helpful to identify exactly what file is having the problem.  Since we're doing containers the file path gets changed to the container env and that's reflected in output.  We need to sed to munge the path back.  In the course of doing so we're losing the return code.
* debug mode ... Makefile echo each rule line by default.  Some of the rules/cmds are really long and make runnning a Makefile target obnoxious.  I like seeing them because we can copy/paste directly into cmdline to debug. I'm seek a happy medium and add a hidden debug mode to trigger verbose rule/cmd output.

**Comment by mazzystr** [comment]
@codificat I'd say we're ready for merge once CI completes

**Comment by mazzystr** [comment]
Stand by for user-guide link fix in another pull req then rebase.

**Comment by codificat** [comment]
With the latest changes, spell checking is now failing for me (IIRC it was working before):

```sh
$ make check_spelling_userguide

Makefile: Check userguide spelling
Error: /scratch/git/kubevirt/user-guide/docs:/srv/docs:Z:ro: incorrect volume format, should be [host-dir:]ctr-dir[:option]
Complete!
$ make check_spelling_website

Makefile: Check website spelling
Error: /scratch/git/kubevirt/website:/srv:Z:ro: incorrect volume format, should be [host-dir:]ctr-dir[:option]
Complete!
```

**Comment by mazzystr** [comment]
@codificat SELinux enforcing/permissive confirmed for check/run/stop/build targets.  This is going to need to be something we should CI/CD in the near future.  It was really painful to test everything.

**Pull request merged**